pkg/kubectl/cmd/resource/get.go

```diff
@@ -584,7 +586,15 @@ func (options *GetOptions) watch(f cmdutil.Factory, cmd *cobra.Command, args []s
 	first := true
 	intr := interrupt.New(nil, w.Stop)
 	intr.Run(func() error {
-		_, err := watch.Until(0, w, func(e watch.Event) (bool, error) {
+		timeout := cmdutil.GetFlagDuration(cmd, "timeout")
+
+		// FIXME: Switch to an informer (wtools.UntilWithInformer)
+		// Pure WATCH will fail after some time but definitely on API timeout.
+		// This is not an easy task as the CLI arguments "watch" and "watch-only"
+		// enforce using pure WATCH even in it's description. Those need to be deprecated first.
```

Comment:
s/it's/its/

Reply:
If we do switch to using an informer, and the watch times out, would you have it do the normal informer behavior or relist (notifying the consumer) and then watching again?

Reply:
I'd choose a regular informer behaviour. Informer will try to run watch again from the last RV if possible, re-list if not. Continuing watch from last RV in case of a apiserver (or similar) timeout seems preferable. That will give user the best chance to see the exact order of events unless that's no longer possible (too old RV) and notifies him when that happens.